tools/generate_report.rb

```diff
@@ -26,7 +26,24 @@ def report_from_args
 puts "Generating report... #{report.name}"
 
 report.queue_generate_table(:userid => USER_ID)
-report._async_generate_table(MiqTask.last.id, REPORT_PARAMS)
+def report_run
+  if defined?(ManageIQPerformance)
+    ManageIQPerformance.profile('generate_report') do
+      total_report_time = Benchmark.measure { yield }
```

Comment:
So I do have a time value (sort of) returned by the `ManageIQPerformance.profile`, but as you discovered, it doesn't have a good way of making that available outside the, or even inside, the block.  I have and idea on how I think we could pass back the data back from the `.profile` method as just a object of the data we have collected, but I will have to work something up for that.  I think that would work well here.